### PR TITLE
Use unitttest discover, fix deprecation warnings in tests

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This uses the suggestions from @jugmac00 in PR #2.

No significant change, so no news snippet.
No real code change, so no Jenkins jobs, only Travis.

### CODE REVIEW

File: plone/synchronize/tests.py [hunk 1]
```diff
@@ -46,14 +46,14 @@ def test_instance_method(self):
         except IndexError:
             raised = True
 
-        self.failUnless(raised)
+        self.assertTrue(raised)
```
Comment (jugmac00) [inline plone/synchronize/tests.py]: I like `self.assertRaises(IndexError, shared_stack.pop)` better, as it is clearer to me, but your solution totally works and is less code changed.

**Review by jugmac00 — APPROVED** [review]
lgtm!

(as a side note I am a pure Zope user, and only signed up for the Plone repositories because of plone.recipe.zope2instance, but I guess these changes are straight forward enough so I can do this review - otherwise some other Plone user should have a look)

File: plone/synchronize/tests.py (hunk 1 above)
Comment (mauritsvanrees) [reply plone/synchronize/tests.py]: Ah, I did not look carefully enough at your patch and just did a quick search-and-replace myself. I will do an extra commit with assertRaises now (used as context manager).

File: plone/synchronize/tests.py [hunk 2]
```diff
@@ -38,15 +38,8 @@ def test_instance_method(self):
 
         self.assertEqual("one", item)
 
-        # raises exception
-
-        raised = False
-        try:
+        with self.assertRaises(IndexError):
             shared_stack.pop()
```
Comment (jugmac00) [inline plone/synchronize/tests.py:42]: Excellent! As an almost 100% pytest user I did not know context managers are available for the unittest module.

Thanks, I learned something new!

**Pull request merged**